### PR TITLE
Adding MSSQL Support

### LINKED ISSUE #22: Add MSSQL Support
Need to wait for MySQL grammar to finish

### PR DESCRIPTION
**Pull request by josephmancuso**
Closes #22

### CODE REVIEW

**Comment by josephmancuso** [comment]
@Marlysson @girardinsamuel this only took me like 3 days to do based on the current data structures but I think its pretty much done. What I have to do now is just double check the tests I'm asserting against a real Microsoft database

File: src/masoniteorm/connections/MSSQLConnection.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+from .BaseConnection import BaseConnection
+from ..query.grammars import MSSQLGrammar
+from ..schema.platforms import MSSQLPlatform
+
+
+CONNECTION_POOL = []
+
+
+class MSSQLConnection(BaseConnection):
+    """Postgres Connection class."""
```
Comment (girardinsamuel) [inline src/masoniteorm/connections/MSSQLConnection.py]: Don't forget to update the docstring here
```suggestion
    """MSSQL Connection class."""
```

File: src/masoniteorm/connections/MSSQLConnection.py [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+        """This sets the connection on the connection class"""
+        try:
+            import pyodbc
+        except ModuleNotFoundError:
+            raise DriverNotFound(
+                "You must have the 'pyodbc' package installed to make a connection to Microsoft SQL Server. Please install it using 'pip install pyodbc'"
+            )
+
+        self._connection = pyodbc.connect(
+            f"DRIVER={'ODBC Driver 17 for SQL Server'};SERVER={self.host},{self.port};DATABASE={self.database};UID={self.user};PWD={self.password}",
```
Comment (girardinsamuel) [inline src/masoniteorm/connections/MSSQLConnection.py]: Shouldn't this hardcoded DRIVER string be defined from config ? I don't know MSSQL very well so might be wrong.. . I guess it's also linked to the version of `pyodbc` we are using ?

File: src/masoniteorm/connections/MSSQLConnection.py [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+    def commit(self):
+        """Transaction"""
+        if self.get_transaction_level() == 1:
+            self._connection.commit()
+            self._connection.autocommit = True
+
+        self.transaction_level -= 1
+
+    def begin(self):
+        """Postgres Transaction"""
```
Comment (girardinsamuel) [inline src/masoniteorm/connections/MSSQLConnection.py]:
```suggestion
        """MSSQL Transaction"""
```

File: src/masoniteorm/schema/platforms/MSSQLPlatform.py [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+        return sql
+
+    def add_column_string(self):
+        return "{name} {data_type}{length}"
+
+    def drop_column_string(self):
+        return "{name}"
+
+    def rename_column_string(self, table, old, new):
+        return f"EXEC sp_rename 'users.post', 'comment', 'COLUMN';"
```
Comment (girardinsamuel) [inline src/masoniteorm/schema/platforms/MSSQLPlatform.py]: Guess this first testing line can be removed
```suggestion
```

**Review by girardinsamuel — COMMENTED** [review]
@josephmancuso Made a quick review 👍

File: src/masoniteorm/schema/platforms/MSSQLPlatform.py (hunk 4 above)
Comment (josephmancuso) [reply src/masoniteorm/schema/platforms/MSSQLPlatform.py]: The first line is correct. the second line needs to be removed :)

File: src/masoniteorm/connections/MSSQLConnection.py (hunk 2 above)
Comment (josephmancuso) [reply src/masoniteorm/connections/MSSQLConnection.py]: You are right. We're gonna need a way to pass all the other options into the connection class. Right now we just pass the host, username, password, etc

**Comment by josephmancuso** [comment]
This at least has feature parity with the other databases so will merge. Will also look into git store to store this mssql driver

**Pull request merged**